### DISCUSSION TITLE
Expected behavior of Keda when IdleReplicaCount is Set or not Set

### DISCUSSION THREAD

**Discussion by aditya-msd** [kedacore/keda#4553] [Q&A / Need Help] [answered]
Want to know the behaviour of Keda for the following scenario , with Scaler being linked to Kafka . 
**Scenario 1** : I have setup a deployment with min replicas = 1 and max replicas = 2 . Additionally have also idle replica count = 0 , so that Keda scales down the deployment to 0 , when there is no traffic or msg in the related Kafka Topic . 

Scenario 2 :  Given that Scenario leads to no pods running when there is no msg in the kafka topic , in this , what I essentially want is to remove idle replica count =0 , so that there will be at the minimum 1 pod that always runs . 

How do I achieve this ? 

I tried removing idleReplicaCount = 0 , from the Scaled Object via API , with the assumption that the keda scalers will scale up the deployment to min replicas , but it seems to not work . 

May be I am understanding of the behaviour of Keda differently.

**Comment by zroubalik** [comment, marked as answer]
Just omit `idleReplicaCount` from the ScaledObject. And set `minReplicaCount` to `1` and max to whatever suits you.

**Reply by aditya-msd** [reply]
I did this . What I observed is that the pods come up based on the minReplicaCount and immediately they are scaled back to 0.

**Reply by zroubalik** [reply]
Can you shared your scaledobject and KEDA operator logs? That behavior should not happen.

**Reply by aditya-msd** [reply]
I reviewed my code to see if I am doing any mistake . Turns out the python kubernetes client was not patching the scaled object properly. If we omit the `idleReplicaCount` , then the patching does not remove . Due to this , the scaling was not working as expected . 
Wanted to confirm , if this has anything to do with Keda or not . 
If not , then Keda is working as expected .

**Reply by zroubalik** [reply]
No, it doesn't :)
